Mathlib/Analysis/Normed/Unbundled/AlgNormOfGalois.lean

```diff
@@ -0,0 +1,143 @@
+/-
+Copyright (c) 2023 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: María Inés de Frutos-Fernández
+-/
+import Mathlib.Analysis.Normed.Unbundled.FiniteExtension
+import Mathlib.LinearAlgebra.FreeModule.Finite.Matrix
+
+/-!
+# algNorm_of_auto and algNorm_of_galois
+
+Let `K` be a nonarchimedean normed field and `L/K` be a finite algebraic extension. In the comments,
+`‖ ⬝ ‖` denotes any power-multiplicative algebra norm on `L` extending the norm  on `K`.
+
+## Main Definitions
+
+* `algNorm_of_auto` : given `σ : L ≃ₐ[K] L`, the function `L → ℝ` sending `x : L` to `‖ σ x ‖` is
+  an algebra norm on `K`.
+* `algNorm_of_galois` : the function `L → ℝ` sending `x : L` to the maximum of `‖ σ x ‖` over
+  all `σ : L ≃ₐ[K] L` is an algebra norm on `L`.
+
+## Main Results
+* `isPowMul_algNorm_of_auto` : `algNorm_of_auto` is power-multiplicative.
+* `isNonarchimedean_algNorm_of_auto` : `algNorm_of_auto` is nonarchimedean.
+* `algNorm_of_auto_extends` : `algNorm_of_auto` extends the norm on `K`.
+* `isPowMul_algNorm_of_galois` : `algNorm_of_galois` is power-multiplicative.
+* `isNonarchimedean_algNorm_of_galois` : `algNorm_of_galois` is nonarchimedean.
+* `algNorm_of_galois_extends` : `algNorm_of_galois` extends the norm on `K`.
+
+## References
+* [S. Bosch, U. Güntzer, R. Remmert, *Non-Archimedean Analysis*][bosch-guntzer-remmert]
+
+## Tags
+
+algNorm_of_auto, algNorm_of_galois, norm, nonarchimedean
+-/
+
+
+
+
+open scoped NNReal
+
+noncomputable section
+
+variable {K : Type*} [NormedField K] {L : Type*} [Field L] [Algebra K L]
+  (h_fin : FiniteDimensional K L)
+section algNorm_of_auto
+
+/-- Given a normed field `K`, a finite algebraic extension `L/K` and `σ : L ≃ₐ[K] L`, the function
+`L → ℝ` sending `x : L` to `‖ σ x ‖`, where `‖ ⬝ ‖` is any power-multiplicative algebra norm on `L`
+extending the norm on `K`, is an algebra norm on `K`. -/
+def algNorm_of_auto (hna : IsNonarchimedean (norm : K → ℝ)) (σ : L ≃ₐ[K] L) :
```

Comment:
```suggestion
def IsNonarchimedean.algNormOfAlgEquiv (hna : IsNonarchimedean (norm : K → ℝ)) (σ : L ≃ₐ[K] L) :
```
I think `auto` isn't part of the naming convention and you're going to have to go with AlgEquiv. It strikes me as a little odd to put this in the root namespace -- maybe put it in `IsNonarchimedean` because the entire theory is developed under this assumption and it seems like there are other theorems in that namespace already. It should definitely all be capitalised (camel case), even the Of (e.g. we have `LinearEquiv.extendScalarsOfSurjective`).